nemo/collections/cv/models/mnist_lenet5.py

```diff
@@ -74,12 +115,42 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
     def forward(self, images):
         return self.module.forward(images=images)
 
+    # This will fail as train_dataloader is not yet instantiated !
+    # def configure_optimizers(self):
+    #     # Get optimizer class.
+    #     optim_config = self._cfg.optim
+    #     optimizer = self.setup_optimization(optim_config)
+    #     scheduler = prepare_lr_scheduler(
+    #         optimizer, scheduler_config=optim_config, train_dataloader=self._train_dataloader
+    #     )
+    #     return [optimizer], [scheduler]
+
+    def setup_optimization(self, optim_params=None) -> Optimizer:
+        optim_params = optim_params or self._cfg.optim
+        self._optimizer = super().setup_optimization(optim_params)
+        self._scheduler = prepare_lr_scheduler(
+            optimizer=self._optimizer, scheduler_config=optim_params, train_dataloader=self._train_dl
+        )
+
+    def setup_training_data(self, train_data_layer_params: Optional[Dict] = None):
```

Comment:
Small nitpick - here instead of Optional[Dict] hint it should be Optional[MNISTDatasetConfig], right?